js/stdlib/ts/ALGO.ts

```diff
@@ -641,21 +651,26 @@ export class EventCache {
         .txType('appl')
 
     if (minRound) {
-      query = query.minRound(minRound + 1);
-    }
-    if (maxRound) {
-      query = query.maxRound(maxRound);
+      // If cache has: [100, 200]
+      // & querying  : [150, 1000]
+      // We already searched cache for [150, 200] so query network for [201, 1000]
+      query = query.minRound(Math.max(this.currentRound, minRound) + 1);
     }
     if (specRound) {
       query = query.round(specRound);
```

Comment:
If you include this, then when you verify the contract, you won't get all of the events that will be needed next, so you'll have to two indexer trips. The `specRound` should just be the lower bound. In other words...